Sources/MessagingInApp/Gist/Managers/MessageManager.swift

```diff
@@ -99,9 +100,22 @@ extension MessageManager: EngineWebDelegate {
         }
     }
 
+    func trackClickedMetric(action: String, name: String) {
+        // a close action does not count as a clicked action.
+        if action != "gist://close" {
+            if let deliveryId = currentMessage.deliveryId {
+                eventBusHandler.postEvent(TrackInAppMetricEvent(deliveryID: deliveryId, event: InAppMetric.clicked.rawValue, params: ["actionName": name, "actionValue": action]))
+            }
+        }
+    }
+
     func tap(name: String, action: String, system: Bool) {
         Logger.instance.info(message: "Action triggered: \(action) with name: \(name)")
-        delegate?.action(message: currentMessage, currentRoute: currentRoute, action: action, name: name)
+        // This condition executes only for modal messages and not inline messages.
+        if currentMessage.isModalMessage {
+            delegate?.action(message: currentMessage, currentRoute: currentRoute, action: action, name: name)
+        }
```

Comment:
I do understand [why the delegate is only being called for modal messages](https://github.com/customerio/customerio-ios/pull/770#discussion_r1681111606). I do see value in conditionally deciding if the `delegate.action` should be called or not. 

I think it would be better if the modal and inline `MessageManager` subclasses decided when the `delegate.action` function is called instead of this logic being inside of the generic `MessageManager` instance. 

As a suggestion, [here is a commit explaining what I mean](https://github.com/customerio/customerio-ios/pull/770#discussion_r1681111606). I especially like how the commit keeps the `delegate.action` call contained in the `MessageManager` and removes the `Gist.shared.action` call from the inline View.